prow/plugins/updateconfig/updateconfig.go

```diff
@@ -290,11 +292,27 @@ func handle(gc githubClient, kc corev1.ConfigMapsGetter, defaultNamespace string
 		if cm.Namespace == "" {
 			cm.Namespace = defaultNamespace
 		}
-		logger := log.WithFields(logrus.Fields{"configmap": map[string]string{"name": cm.Name, "namespace": cm.Namespace}})
-		if err := Update(&gitHubFileGetter{org: org, repo: repo, commit: *pr.MergeSHA, client: gc}, kc.ConfigMaps(cm.Namespace), cm.Name, cm.Namespace, data, metrics, logger); err != nil {
+		if cm.Cluster == "" {
+			cm.Cluster = kube.DefaultClusterAlias
+		}
+		logger := log.WithFields(logrus.Fields{"configmap": map[string]string{"name": cm.Name, "namespace": cm.Namespace, "cluster": cm.Cluster}})
```

Comment:
All the following logic seems duplicated - DRY it out, please

Reply:
Done.